test/integration/vitesscluster/topology_test.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2020 PlanetScale Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vitesscluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+	"vitess.io/vitess/go/vt/topo"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/test/integration/framework"
+)
+
+func TestTopology(t *testing.T) {
+	ctx := context.Background()
+
+	f := framework.NewFixture(ctx, t)
+	defer f.TearDown()
+
+	ns := "default"
+	cluster := "test-topology"
+
+	// Start with the same basic VitessCluster as the regular test.
+	vt := &planetscalev2.VitessCluster{}
+	framework.MustDecodeYAML(basicVitessCluster, vt)
+
+	// Set the cluster to use an external lockserver, namely the same etcd that
+	// we use for our test k8s apiserver.
+	//
+	// This lets us actually test the operator's topology interaction.
+	// Otherwise, there is no actual etcd since this integration test
+	// environment doesn't actually run any Pods.
+	params := &planetscalev2.VitessLockserverParams{
+		Implementation: "etcd2",
+		Address:        framework.EtcdURL(),
+		RootPath:       fmt.Sprintf("/vitess/%s/", cluster),
```

Comment:
In case others want to add more paths onto the `RootPath`, would it be better to not have a tail forward slash? I'm just thinking about the format of an addition to the path. It would require that any additions do *not* start with a forward slash. Thoughts?

Reply:
The idiomatic way to join paths in Go is `path.Join()` (for abstract paths) or `filepath.Join()` (for filesystem paths) which ensures separators are consistent regardless of input. It's also a good practice to design interfaces that accept paths to work either with or without a trailing slash whenever possible. So if it matters what people give as input, we need to fix the receiver rather than the input.

Reply:
Good to know, thanks!